util.go

```diff
@@ -280,12 +283,16 @@ func decompressPayload(msg []byte) ([]byte, error) {
 func decompressBuffer(c *compress) ([]byte, error) {
 	// Verify the algorithm
 	if c.Algo != lzwAlgo {
-		return nil, fmt.Errorf("Cannot decompress unknown algorithm %d", c.Algo)
+		return nil, fmt.Errorf("cannot decompress unknown algorithm %d", c.Algo)
 	}
 
 	// Create a uncompressor
 	uncomp := lzw.NewReader(bytes.NewReader(c.Buf), lzw.LSB, lzwLitWidth)
-	defer uncomp.Close()
+	defer func() {
+		if err := uncomp.Close(); err != nil {
+			log.Fatal(err)
+		}
+	}()
```

Comment:
log.Fatal calls os.Exit, which is an extremely problematic side-effect.
Swallow the error instead.